cms/djangoapps/contentstore/api/views/course_validation.py

```diff
@@ -207,5 +209,63 @@ def _get_assignments(self, course):
         ]
         return assignments, visible_assignments
 
+    def _get_open_responses(self, assignment):
+        store = modulestore()
+        verticals = [
+            store.get_item(vertical_usage_key)
+            for vertical_usage_key in assignment.children
+        ]
+        oras = [
+            store.get_item(item_usage_key)
+            for vertical in verticals
+            for item_usage_key in vertical.children
+            if 'type@openassessment' in str(item_usage_key)
+        ]
```

Comment:
Modulestore provides an interface to find all blocks of a certain type:

```
modulestore().get_items(course.id, qualifiers={'category': 'openassessment'})
```

This will likely be more efficient.  You should be able to use this method if you separate the ORA payload from the other assignments.